mitmproxy/tools/console/master.py

```diff
@@ -438,6 +438,18 @@ def clear_events(self):
 
     # Handlers
     @controller.handler
+    def websocket_message(self, f):
+        super().websocket_message(f)
+        message = f.messages[-1]
+        direction = "->" if message.from_client else "<-"
+        self.add_log("{client} {direction} websocket {direction} {server}".format(
```

Comment:
After we add the handshake to the flow object, this should show the endpoint URL used to establish the websocket connection.